src/platform/testing/contract/client.js

```diff
@@ -0,0 +1,303 @@
+
+    const url = urlString(
+      path.join(
+        this.url,
+        'pacts',
+        `provider/${provider}`,
+        `consumer/${consumer}`,
+        `version/${version}`,
+      ),
+    );
+
+    const response = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        ...this.authHeader,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (response.ok) {
+      console.log(
+        `Published pact between consumer ${consumer} and provider ${provider}.`,
+      );
+    } else {
+      throw new Error(
+        `Failed to publish pact between consumer ${consumer} and provider ${provider}: ${
+          response.status
+        } ${response.statusText}`,
+      );
+    }
+
+    return pactObject;
+  };
+
+  /**
+   * Tags a pact.
+   * @params {PactObject} pactObject
+   */
+  tagPact = async pactObject => {
+    const { data, version, tag } = pactObject;
+    const consumer = data.consumer.name;
+
+    const url = urlString(
+      path.join(
+        this.url,
+        `pacticipants/${consumer}`,
+        `versions/${version}`,
+        `tags/${tag}`,
+      ),
+    );
+
+    const response = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        ...this.authHeader,
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (response.ok) {
+      console.log(`Tagged version ${version} of ${consumer} as "${tag}".`);
+    } else {
+      throw new Error(
+        `Failed to tag version ${version} of consumer ${consumer} as "${tag}": ${
+          response.status
+        } ${response.statusText}`,
+      );
+    }
+
+    return pactObject;
+  };
+
+  /**
+   * Publishes and tags a pact.
+   * @param {PactObject} pactObject
+   */
+  publishAndTagPact = pactObject =>
+    Promise.resolve(pactObject)
+      .then(this.publishPact)
+      .then(this.tagPact);
+
+  /**
+   * Publishes and tags all pacts in the pacts folder.
+   * @param {BrokerOperationOptions} options
+   */
+  publishAndTagPacts = async options => {
+    try {
+      validateRequiredOptions(
+        'publishAndTagPacts',
+        ['pactDir', 'version', 'tag'],
+        options,
+      );
+
+      const { pactDir, version, tag } = options;
+
+      const pactObjects = this.parsePacts(pactDir).map(data => ({
+        data,
+        version,
+        tag,
+      }));
+
+      await Promise.all(pactObjects.map(this.publishAndTagPact));
+      console.log('Successfully published and tagged pacts.');
+    } catch (e) {
+      console.error('Failed to publish and tag pacts.');
+      throw e;
+    }
+  };
+
+  /**
+   * Creates a set of objects that provide context for figuring out
+   * whether the consumers from the generated pacts have passed
+   * verification and can be deployed to the desired environment.
+   * @param {BrokerOperationOptions} options
+   * @return {VerificationContext[]}
+   */
+  createVerificationContexts = ({ pactDir, version, tag }) => {
```

Comment:
Mainly used to construct the URL for figuring out `can-i-deploy` from the Pact Broker API.